prefect_azure/container_instance.py

```diff
@@ -0,0 +1,704 @@
+    ) -> ContainerInstanceJobResult:
+        """
+        Runs the configured task using an ACI container.
+
+        Args:
+            task_status: An optional `TaskStatus` to update when the container starts.
+
+        Returns:
+            An `ContainerInstanceJobResult` with the container's exit code.
+        """
+
+        run_start_time = datetime.datetime.now(datetime.timezone.utc)
+        token_credential = self._create_credential()
+        aci_client = self._create_container_client(token_credential)
+        container = self._configure_container()
+        container_group = self._configure_container_group(token_credential, container)
+        created_container_group = None
+
+        self.logger.info(
+            f"{self._log_prefix}: Preparing to run command {' '.join(self.command)!r} "
+            f"in container {container.name!r} ({self.image})..."
+        )
+        try:
+            self.logger.info(f"{self._log_prefix}: Waiting for container creation...")
+            # Create the container group and wait for it to start
+            creation_status_poller = aci_client.container_groups.begin_create_or_update(
+                self.resource_group_name, container.name, container_group
+            )
+            created_container_group = await run_sync_in_worker_thread(
+                self._wait_for_task_container_start, creation_status_poller
+            )
+
+            # If creation succeeded, group provisioning state should be 'Succeeded'
+            # and the group should have a single container
+            if self._provisioning_succeeded(created_container_group):
+                self.logger.info(f"{self._log_prefix}: Running command...")
+                if task_status:
+                    task_status.started(value=container.name)
+                status_code = await run_sync_in_worker_thread(
+                    self._watch_task_and_get_exit_code,
+                    aci_client,
+                    created_container_group,
+                    run_start_time,
+                )
+                self.logger.info(f"{self._log_prefix}: Completed command run.")
+            else:
+                raise RuntimeError(f"{self._log_prefix}: Container creation failed.")
+
+        finally:
+            if created_container_group:
+                self.logger.info(f"{self._log_prefix}: Deleting container...")
+                aci_client.container_groups.begin_delete(
+                    resource_group_name=self.resource_group_name,
+                    container_group_name=created_container_group.name,
+                )
+
+        return ContainerInstanceJobResult(
+            identifier=container.name, status_code=status_code
+        )
+
+    def preview(self) -> str:
+        """
+        Provides a summary of how the container will be created when `run` is called.
+
+        Returns:
+           A string containing the summary.
+        """
+        preview = {
```

Comment:
The preview should be able contain generated information too, as long as it doesn't require calls to the actual service e.g. I think the output of `_configure_container` could probably be in here? The more information we can give the user about what we're generating the better they can debug their configuration.

This seems like a large change. I'd be happy to see it tracked in an issue / follow-up to get this merged asap.

Reply:
Opened an issue here https://github.com/PrefectHQ/prefect-azure/issues/48